[ko] Update content/ko/docs/concepts/services-networking/service.md (#56714)
kubernetes/website: content/ko/docs/concepts/services-networking/service.md

Reviewer: ```suggestion
하위 엔드포인트 1000개 제한은 레거시 엔드포인트 API에만 적용된다.
```

문서 내 영단어 `legacy`에 대해 `기존`과` 레거시`가 혼용되어있으므로 통일 부탁드립니다.

Reviewer: ```suggestion
    쿠버네티스를 사용하면 익숙하지 않은 서비스 디스커버리 메커니즘을 사용하기 위해 애플리케이션을 수정할 필요가 없다. 쿠버네티스는 파드에게 고유한 IP 주소와 파드 집합에 대한 단일 DNS 명을 부여하고, 그것들 간에 로드 밸런스를 수행할 수 있다.
```

[리뷰 참고](https://github.com/kubernetes/website/pull/56714/changes#r3703655521)

<!-- overview -->

{{< glossary_definition term_id="service" length="short" prepend="쿠버네티스에서 서비스는" >}}

쿠버네티스에서 서비스의 주된 목적은 익숙하지 않은 서비스 디스커버리 메커니즘을 사용하기
위해 기존 애플리케이션을 수정할 필요가 없도록 하는 것이다.

> 클라우드 네이티브 환경에 맞게 설계된 코드이든, 컨테이너화된 오래된 애플리케이션이든

Author: 리뷰 내용을 반영하여 이 부분도 `클라우드-네이티브`에서 `클라우드 네이티브`로 수정했습니다.

> {{< glossary_definition term_id="service" length="short" >}}

Reviewer: ```suggestion
{{< glossary_definition term_id="service" length="short" prepend="쿠버네티스에서 서비스는" >}}
```
> {{< glossary_definition term_id="service" length="short" prepend="In Kubernetes, a Service is" >}}

`prepend`는 glossary_definition으로 가져오는 용어 정의 앞에 특정 문구를 덧붙이는 옵션입니다. 
따라서 일반 문장과 마찬가지로 그대로 한글화해주시면 됩니다.

쿠버네티스에서 서비스는 익숙하지 않은 서비스 디스커버리 메커니즘을 사용하기
위해 기존 애플리케이션을 수정할 필요가 없도록 하는 것이다.
클라우드-네이티브 환경에 맞게 설계된 코드이든, 컨테이너화된 오래된 애플리케이션이든
파드에서 실행할 수 있다. 클라이언트가 상호작용할 수 있도록, 서비스를 사용하여 해당

> 파드 집합을 네트워크에서 사용할 수 있게 한다.

Reviewer: ```suggestion
쿠버네티스에서 서비스의 주된 목적은 익숙하지 않은 서비스 디스커버리 메커니즘을 사용하기
위해 기존 애플리케이션을 수정할 필요가 없도록 하는 것이다.
클라우드-네이티브 환경에 맞게 설계된 코드이든, 컨테이너화된 오래된 애플리케이션이든
파드에서 실행할 수 있다. 서비스를 사용하면 파드 집합을 네트워크에서 사용할 수 있게 되어
클라이언트가 이를 통해 상호 작용할 수 있다.
```

> A key aim of Services in Kubernetes is that you don't need to modify your existing
application to use an unfamiliar service discovery mechanism.
You can run code in Pods, whether this is a code designed for a cloud-native world, or
an older app you've containerized. You use a Service to make that set of Pods available
on the network so that clients can interact with it.

단어 생략(`A key aim`) 없이 보다 자연스럽게 바꾸어 제안드립니다.

애플리케이션을 실행하기 위해 {{< glossary_tooltip term_id="deployment" >}}를 사용하는 경우, 해당
디플로이먼트는 파드를 동적으로 생성하고 제거할 수 있다. 어느 한 순간에서 다음 순간으로 넘어갈 때,
그 파드들 중 몇 개가 동작하고 정상 상태인지 알 수 없으며, 심지어 정상 상태인
파드의 이름이 무엇인지조차 알 수 없다.
쿠버네티스 {{< glossary_tooltip term_id="pod" text="파드" >}}는 클러스터의 목표 상태와 일치하도록

> 생성되고 삭제된다. 파드는 비영구적 리소스이다(개별 파드가 신뢰할 수

Reviewer: ```suggestion
생성되고 삭제된다. 파드는 비영구적인 리소스이다(개별 파드가 신뢰할 수
```

_서비스_ 로 들어가보자.

<!-- body -->

> ## 서비스 리소스 {#service-resource}

Reviewer: ```suggestion
## 쿠버네티스에서의 서비스
```

> Services in Kubernetes
영어 원문에 맞춰 수정하였습니다.

쿠버네티스의 일부인 서비스 API는 파드 그룹을 네트워크로 노출하는 데 도움이 되는
추상화 개념이다. 각 서비스 오브젝트는 엔드포인트의 논리적

> 집합과 (일반적으로 이러한 엔드포인트는 파드이다), 해당 파드에 접근하는 방법에 대한 정책을 정의한다.

Reviewer: ```suggestion
쿠버네티스의 일부인 서비스 API는 파드 그룹을 네트워크를 통해 노출할 수 있도록 도와주는
추상화이다. 각 서비스 오브젝트는 엔드포인트의 논리적
집합(일반적으로 이러한 엔드포인트는 파드이다)과, 해당 파드에 접근할 수 있도록 하는 정책을 함께 정의한다.
```

> The Service API, part of Kubernetes, is an abstraction to help you expose groups of
Pods over a network. Each Service object defines a logical set of endpoints (usually
these endpoints are Pods) along with a policy about how to make those pods accessible.

원문에 맞게 조금 다듬어 보았습니다.

서비스 추상화는 이러한 디커플링을 가능하게 한다.

서비스가 대상으로 하는 파드 집합은 일반적으로 사용자가 정의하는
{{< glossary_tooltip text="셀렉터" term_id="selector" >}}에 의해
결정된다.
서비스 엔드포인트를 정의하는 다른 방법에 대해 알아보려면,
[셀렉터가 _없는_ 서비스](#셀렉터가-없는-서비스)를 참고한다.

워크로드가 HTTP를 사용한다면, 웹 트래픽이 해당 워크로드에 도달하는
방식을 제어하기 위해
[인그레스(Ingress)](/docs/concepts/services-networking/ingress/)를 사용하기로 선택할 수도 있다.
인그레스는 서비스 유형이 아니지만, 클러스터의 진입점 역할을 한다. 인그레스를
사용하면 라우팅 규칙을 단일 리소스로 통합할 수 있으므로, 클러스터 내에서 개별적으로
실행되는 워크로드의 여러 컴포넌트를 단일 리스너 뒤에서 노출할 수
있다.

쿠버네티스용 [게이트웨이(Gateway)](https://gateway-api.sigs.k8s.io/#what-is-the-gateway-api)
API는 인그레스와 서비스를 넘어서는 추가 기능을 제공한다. 게이트웨이는 클러스터에
추가할 수 있으며, {{< glossary_tooltip term_id="CustomResourceDefinition" text="커스텀리소스데피니션" >}}을
사용하여 구현되는 익스텐션(extension) API 집합이다 -

> 이후 이를 사용하여 클러스터에서 실행 중인 네트워크 서비스에 대한 접근을 구성할 수 있다.

Reviewer: ```suggestion
쿠버네티스용 [게이트웨이](https://gateway-api.sigs.k8s.io/#what-is-the-gateway-api)
API는 인그레스와 서비스를 넘어서는 추가 기능을 제공한다. 게이트웨이는
{{< glossary_tooltip term_id="CustomResourceDefinition" text="커스텀리소스데피니션" >}}을
사용하여 구현된 확장 API 집합으로, 클러스터에 추가한 후 이를 사용하여
클러스터에서 실행 중인 네트워크 서비스에 대한 접근을 구성할 수 있다.
```

`게이트웨이 API`로 표현하는 것이 더 자연스러울 것 같은데 어떠신지요?

--- 

> The [Gateway](https://gateway-api.sigs.k8s.io/#what-is-the-gateway-api) API for Kubernetes
provides extra capabilities beyond Ingress and Service. You can add Gateway to your cluster -
it is a family of extension APIs, implemented using
{{< glossary_tooltip term_id="CustomResourceDefinition" text="CustomResourceDefinitions" >}} -
and then use these to configure access to network services that are running in your cluster.

한글화 팀에서는 영어 원문에서 사용된 문장 부호 중 한국어 문장에 그대로 적용할 수 없는 경우에는 한국어 문장 부호 체계에 맞게 조정하고 있습니다. 이에 맞춰 수정 제안드립니다.

[참고]
[한글화 모범사례 문장 부호 관련 내용](https://kubernetes.io/ko/docs/contribute/localization_ko_best_practice/#:~:text=%EB%AC%B8%EC%B2%B4%20%EB%B0%8F%20%ED%91%9C%ED%98%84-,%ED%95%9C%EA%B5%AD%EC%96%B4%20%EB%AC%B8%EC%9E%A5%20%EB%B6%80%ED%98%B8%20%EC%B2%B4%EA%B3%84%EC%97%90%20%EB%A7%9E%EA%B2%8C%20%EB%B3%80%ED%99%98%ED%95%98%EC%97%AC%20%EC%82%AC%EC%9A%A9%ED%95%A8,-%EC%98%81%EC%96%B4%20%EC%9B%90%EB%AC%B8%EC%97%90%EC%84%9C%20%EC%82%AC%EC%9A%A9%EB%90%9C)

> ### 클라우드-네이티브 서비스 디스커버리

Reviewer: ```suggestion
### 클라우드 네이티브 서비스 디스커버리
```

Reviewer: [리뷰 참고](https://github.com/kubernetes/website/pull/56714/changes#r3703655521)

네이티브 애플리케이션이 아닌 (non-native applications) 경우, 쿠버네티스는 애플리케이션과 백엔드 파드 사이에 네트워크 포트 또는 로드
밸런서를 배치할 수 있는 방법을 제공한다.

어느 쪽이든, 워크로드는 연결하려는 대상을 찾기 위해 이러한
[서비스 디스커버리](#서비스-디스커버리하기) 메커니즘을 사용할 수 있다.

## 서비스 정의

서비스는 {{< glossary_tooltip text="오브젝트" term_id="object" >}}이다(파드나 컨피그맵이 오브젝트인 것과 마찬가지로).
쿠버네티스 API를 사용하여 서비스
정의를 생성, 조회 또는 수정할 수 있다. 일반적으로는 

> `kubectl`과 같은 도구를 사용하여 이러한 API 호출을 대신 수행한다.

Reviewer: ```suggestion
`kubectl`과 같은 도구를 사용하여 이러한 API 호출을 수행한다.
```

> Usually you use a tool such as `kubectl` to make those API calls for you.

`대신`의 주체가 다소 모호하게 읽힐 수 있다는 생각이 듭니다.

예를 들어, 각각 TCP 포트 9376에서 수신하고
`app.kubernetes.io/name=MyApp` 레이블을 가지고 있는 파드 세트가 있다고 가정해 보자.
해당 TCP 리스너를 노출하는 서비스를 다음과 같이 정의할 수 있다.

{{% code_sample file="service/simple-service.yaml" %}}

이 매니페스트를 적용하면 기본 ClusterIP [서비스 유형](#publishing-services-service-types)을
가진 "my-service"라는 새로운 서비스가 생성된다. 이 서비스는 `app.kubernetes.io/name: MyApp`
레이블을 가진 모든 파드의 TCP 포트 9376을 대상으로 한다.

쿠버네티스는 이 서비스에 가상 IP 주소 메커니즘에서 사용되는 IP 주소
(_클러스터 IP_)를 할당한다. 해당 메커니즘에 대한 자세한 내용은
[가상 IP 및 서비스 프록시](/docs/reference/networking/virtual-ips/)를 참고한다.

해당 서비스의 컨트롤러는 셀렉터와 일치하는 파드를 지속적으로 검색하고,
필요에 따라 해당 서비스의 엔드포인트슬라이스 집합에 대한
업데이트를 수행한다.

서비스 오브젝트의 이름은 유효한

> [RFC 1123 레이블 이름](/docs/concepts/overview/working-with-objects/names/#dns-label-names)이어야 한다.

Reviewer: ```suggestion
[RFC 1123 레이블 이름](/docs/concepts/overview/working-with-objects/names#rfc-1123-label-names)이어야 한다.
```

> The name of a Service object must be a valid
[RFC 1123 label name]((/docs/concepts/overview/working-with-objects/names#rfc-1123-label-names).

원문에 맞춰 변경하였습니다.

    ports:
      - containerPort: 80
        name: http-web-svc
```

이것은 서로 다른 포트 번호를 통해 가용한 동일 네트워크 프로토콜이 있고, 
단일 구성 이름을 사용하는 서비스 내에 혼합된 파드가 존재해도 가능하다.
이를 통해 서비스를 배포하고 진전시키는 데 많은 유연성을 제공한다.
예를 들어, 클라이언트를 망가뜨리지 않고,
백엔드 소프트웨어의 다음 버전에서 파드가 노출시키는 포트 번호를 변경할 수 있다.

서비스의 기본 프로토콜은 
[TCP](/docs/reference/networking/service-protocols/#protocol-tcp)이다. 
다른 [지원되는 프로토콜](/docs/reference/networking/service-protocols/)을 사용할 수도 있다.

많은 서비스가 하나 이상의 포트를 노출해야 하기 때문에, 쿠버네티스는 서비스 오브젝트에서
[다중 포트 정의](#멀티-포트-서비스)를 지원한다.
각 포트는 동일한 `프로토콜` 또는 다른 프로토콜로 정의될 수 있다.

### 셀렉터가 없는 서비스

서비스는 일반적으로 셀렉터를 이용하여 쿠버네티스 파드에 대한 접근을 추상화하지만, 

> 셀렉터 대신 대응되는 

Reviewer 2: 해당 문서 내에서 `corresponding`에 대해 3가지의 표현이 혼용되고 있습니다.

1. 186 번 라인 - "대응되는"
2. 216 번 라인 - "매칭되는"
3. 353 번 라인 - "상응하는"

문서 일관성 측면에서 표현 통일을 제안 드립니다 😀

Author: 좋은 리뷰 감사드립니다! 다른 페이지의 선례를 참고하여 해당 용어를 "상응하는"으로 통일했습니다.
186번 라인의 경우, 무엇에 상응하는 건지 명시하는 것이 좋을 것 같아 "서비스에 상응하는"으로 수정했습니다!

  name: my-service
spec:
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: 9376
```

이 서비스에는 셀렉터가 없으므로, 매칭되는 엔드포인트슬라이스 
오브젝트가 자동으로 생성되지 않는다. 
엔드포인트슬라이스 오브젝트를 수동으로 추가하여, 

> 서비스를 실행 중인 네트워크 주소 및 포트에 서비스를 수동으로 매핑할 수 있다. 예시는 다음과 같다.

Reviewer 3: 해당 문장에서 '수동으로', '서비스를' 단어 중복을 제거하는거 어떨까요?

셀럭터가 없는 서비스에 엔드포인트슬라이스를 추가하여 네트워크 주소 및 포트를 매칭할 수 있음을 명확하게 표현할 수 있을거 생각합니다.

Author: @WAFriend3416 
안녕하세요, 리뷰해주셔서 정말 감사드립니다!

다만 해당 부분은 원문과 비교해봤을 때, 번역의 방향성에 대한 논의가 필요할 것 같아 수정 보류 중에 있습니다.

따라서 아래의 논의 사항을 해결하고, 승환님께서 리뷰해주신 내용을 참고하여 수정을 진행하겠습니다!
(승환님도 아래 논의 내용에 대한 의견을 공유해주시면 좋을 것 같습니다!)

논의 내용은 다음과 같습니다. 

> You can map the Service to the network address and port where it's running, by adding an EndpointSlice object manually.

위 영어 원문에서 `it`은 문법적으로 "쿠버네티스의 Service 오브젝트"를 의미하는 것으로 보입니다. 하지만 이 문장을 번역으로 옮길 때 `it`을 Service 오브젝트로 번역하게 된다면 "서비스(오브젝트)가 실행 중"이라는 개념적으로 어색한 문장이 될 것 같아, 해당 Service를 "애플리케이션 서비스"의 의미로 번역하는 것이 더 적절해보입니다. 의미를 풀어서 번역하게 된다면 개념은 명확해지지만 원문을 준수하지 않게 되는 것 같아, 해당 문장을 어떻게 번역하는 것이 좋을지에 대한 의견을 나누어보고 싶습니다!

@wonyongg

> 서비스를 실행 중인 네트워크 주소 및 포트에 서비스를 수동으로 매핑할 수 있다. 예시는 다음과 같다.

Author: @wonyongg 
안녕하세요, 원용님. 해당 라인을 리뷰해주신 @WAFriend3416 님과 오프라인으로 의견을 나누는 과정에서 번역의 방향성에 대한 논의가 필요할 것 같아 멘션드립니다!

논의 내용은 다음과 같습니다. 

> You can map the Service to the network address and port where it's running, by adding an EndpointSlice object manually.

위 영어 원문에서 `it`은 문법적으로 "쿠버네티스의 Service 오브젝트"를 의미하는 것으로 보입니다. 하지만 이 문장을 번역으로 옮길 때 `it`을 Service 오브젝트로 번역하게 된다면 "서비스(오브젝트)가 실행 중"이라는 개념적으로 어색한 문장이 될 것 같아, 해당 Service를 "애플리케이션 서비스"의 의미로 번역하는 것이 더 적절해보입니다. 이와 같이 의미를 풀어서 번역하게 된다면 개념은 명확해지지만 원문을 준수하지 않게 되는 것 같아, 해당 문장을 어떻게 번역하는 것이 좋을지에 대한 의견을 나누어보고 싶습니다!

Reviewer 4: @llokr1 Feel free to enhance via further PR if needed.

  name: my-service
spec:
  ports:
    - name: http
      protocol: TCP
      port: 80
      targetPort: 9376
```

이 서비스에는 셀렉터가 없으므로, 상응하는 엔드포인트슬라이스 
오브젝트가 자동으로 생성되지 않는다. 
엔드포인트슬라이스 오브젝트를 수동으로 추가하여, 

> 서비스가 구동 중인 네트워크 주소 및 포트에 서비스를 수동으로 매핑할 수 있다. 예시는 다음과 같다.

Author: @wonyongg 
2026년 8월 13일 한글화팀 정기회의에서 공유해주신 조언을 바탕으로 '서비스가 구동 중인'이라고 번역했습니다.
해당 변경 사항에 대해서도 보완하거나 더 자연스러운 표현이 있다면 조언해 주시면 감사하겠습니다!

Reviewer: @llokr1 

다시 한 번 자세히 살펴보았을 때 `서비스가 구동 중인`으로 변경하는 것이 가장 적절하고, 말씀하신 부분을 해소할 수 있는 번역인 것 같습니다.

{{< feature-state for_k8s_version="v1.20" state="stable" >}}

`appProtocol` 필드는 각 서비스 포트에 대한 애플리케이션 프로토콜을 지정하는 방법을 제공한다.
이는 구현체가 이해하는 프로토콜에 대해 더 풍부한 동작을
제공하도록 하는 단서로 사용된다.
이 필드의 값은 상응하는 엔드포인트와 엔드포인트슬라이스
오브젝트에 의해 미러링된다.

이 필드는 표준 쿠버네티스 레이블 구문을 따른다. 유효한 값은 다음 중 하나이다.

* [IANA 표준 서비스 이름](https://www.iana.org/assignments/service-names).

* `mycompany.com/my-custom-protocol`과 같이 구현체가 정의한 접두사가 붙은 이름.

* 쿠버네티스가 정의한 접두사가 붙은 이름.

| 프로토콜 | 설명 |
|----------|-------------|
| `kubernetes.io/h2c` | [RFC 9113](https://www.rfc-editor.org/rfc/rfc9113)에 설명된 대로 평문(cleartext)을 통한 HTTP/2 |
| `kubernetes.io/ws`  | [RFC 6455](https://www.rfc-editor.org/rfc/rfc6455)에 설명된 대로 평문을 통한 웹소켓 |
| `kubernetes.io/wss` | [RFC 6455](https://www.rfc-editor.org/rfc/rfc6455)에 설명된 대로 TLS를 통한 웹소켓 |

> ### 멀티-포트 서비스

Reviewer: ```suggestion
### 멀티 포트 서비스
```

[리뷰 참고](https://github.com/kubernetes/website/pull/56714/changes#r3703655521)

소문자 영숫자와 `-` 만 포함해야 한다. 포트 이름은
영숫자로 시작하고 끝나야 한다.

예를 들어, `123-abc`와 `web`은 유효하지만, `123_abc`와 `-web`은 유효하지 않다.
{{< /note >}}

> ## 서비스 유형  {#publishing-services-service-types}

Reviewer: ```suggestion
## 서비스 타입  {#publishing-services-service-types}
```

필드 값이 `type`으로 명시되어있기 때문에 `타입`을 사용하는 것이 가독성 측면에서 더 좋지 않을까 생각이 듭니다.
문서 내에 영문이 `type`인 부분을 모두 `유형`에서 `타입`으로 변경 부탁드립니다.

서비스 대신 [게이트웨이](https://gateway-api.sigs.k8s.io/)와 통합하거나, 동등한 세부
정보를 지정하는 (공급자별) 자체 어노테이션을 서비스에 정의할 수 있다.
{{< /note >}}

#### 로드 밸런서 트래픽에 대한 노드 활성 상태(liveness)의 영향

로드 밸런서 헬스 체크는 최신 애플리케이션에 매우 중요하다. 이는 로드
밸런서가 트래픽을 전달할 서버(가상 머신 또는 IP 주소)를 결정하는 데
사용된다. 쿠버네티스 API는 쿠버네티스가 관리하는 로드 밸런서에 대해
헬스 체크가 어떻게 구현되어야 하는지 정의하지 않으며, 대신 클라우드
공급자(및 통합 코드를 구현하는 사람들)가 그 동작을 결정한다. 로드 밸런서
헬스 체크는 서비스의 `externalTrafficPolicy` 필드를 지원하는 맥락에서
광범위하게 사용된다.

#### 프로토콜 유형이 혼합된 로드 밸런서

{{< feature-state feature_gate_name="MixedProtocolLBService" >}}

기본적으로, LoadBalancer 유형의 서비스에서 둘 이상의 포트가 정의된 경우, 모든 
포트는 동일한 프로토콜을 가져야 하며 프로토콜은 클라우드 공급자가
지원하는 프로토콜이어야 한다.

`MixedProtocolLBService` 기능 게이트(v1.24에서 kube-apiserver에 대해 기본적으로 활성화되어 있음)는 
둘 이상의 포트가 정의되어 있는 경우에 로드 밸런서 유형의 서비스에 대해 서로 다른 프로토콜을 사용할 수 있도록 해 준다.

{{< note >}}
로드 밸런싱되는 서비스에 사용할 수 있는 프로토콜 집합은 클라우드 공급자가 정의한다.
클라우드 공급자는 쿠버네티스 API가 강제하는 것 이상의 제약을 둘 수 있다.
{{< /note >}}

#### 로드 밸런서 NodePort 할당 비활성화 {#load-balancer-nodeport-allocation}

{{< feature-state for_k8s_version="v1.24" state="stable" >}}

`spec.allocateLoadBalancerNodePorts` 필드를 `false`로 설정하여, `type: LoadBalancer`
서비스에 대한 노드 포트 할당을 선택적으로 비활성화할 수 있다. 이는 
노드 포트를 사용하지 않고 트래픽을 파드로 직접 라우팅하는 로드 밸런서 구현에만 사용해야 한다.

> 기본적으로 `spec.allocateLoadBalancerNodePorts`는 `true`이며 LoadBalancer 유형의 서비스는 계속해서 노드 포트를 할당할 것이다. 

Reviewer 2: 맨 끝에 공백이 하나 추가되어 있습니다!

Author: 감사합니다, 수정했습니다!
